src/main/java/org/schedulesdirect/api/Program.java

```diff
@@ -687,6 +708,8 @@ else if(o1 == null && o2 == null)
 			String orig = src.optString("originalAirDate", "");
 			originalAirDate = orig.length() > 0 && !orig.startsWith("0") ? getOrigAirDateFormat().parse(src.getString("originalAirDate")) : null;
 			descriptionLanguage = descs != null ? src.optString("descriptionLanguage", null) : null;
+			String entityTypeStr = src.optString("entityType");
+			entityType = entityTypeStr.length() == 0 ? EntityType.UNKNOWN : EntityType.valueOf(entityTypeStr);
```

Comment:
So here I'm passing the entityTypeStr.toUpperCase() to valueOf() and making sure all vals of the enum are upper case as per Java conventions.